src/app/qgsdxfexportdialog.cpp

```diff
@@ -800,6 +808,239 @@ void QgsDxfExportDialog::deselectDataDefinedBlocks()
+  {
+    outputFileName += QStringLiteral( ".xml" );
+  }
+
+  QString myErrorMessage;
+  QDomDocument myDocument;
+
+  saveSettingsToXML( myDocument );
+
+  const QFileInfo myFileInfo( outputFileName );
+  const QFileInfo myDirInfo( myFileInfo.path() );  //excludes file name
+  if ( !myDirInfo.isWritable() )
+  {
+    QMessageBox::information( this, tr( "Save DXF settings" ), tr( "The directory containing your dataset needs to be writable!" ) );
+    return;
+  }
+
+  QFile myFile( outputFileName );
+  if ( myFile.open( QFile::WriteOnly | QFile::Truncate ) )
+  {
+    QTextStream myFileStream( &myFile );
+    // save as utf-8 with 2 spaces for indents
+    myDocument.save( myFileStream, 2 );
+    myFile.close();
+    QMessageBox::information( this, tr( "Save DXF settings" ), tr( "Created DXF settings file as %1" ).arg( outputFileName ) );
+    settings.setValue( QStringLiteral( "dxf/lastSettingsDir" ), QFileInfo( outputFileName ).absolutePath() );
+    return;
+  }
+  else
+  {
+    QMessageBox::information( this, tr( "Save DXF settings" ), tr( "ERROR: Failed to created DXF Export settings file as %1. Check file permissions and retry." ).arg( outputFileName ) );
+    return;
+  }
+}
+
+
+void QgsDxfExportDialog::saveSettingsToXML( QDomDocument &doc ) const
+{
+  QDomImplementation DomImplementation;
+  const QDomDocumentType documentType = DomImplementation.createDocumentType( QStringLiteral( "qgis" ), QStringLiteral( "http://mrcc.com/qgis.dtd" ), QStringLiteral( "SYSTEM" ) );
+  QDomDocument myDocument( documentType );
+
+  QDomElement myRootNode = myDocument.createElement( QStringLiteral( "qgis" ) );
+  myRootNode.setAttribute( QStringLiteral( "version" ), Qgis::version() );
+  myDocument.appendChild( myRootNode );
+
+  QDomElement symbologyModeElement = myDocument.createElement( QStringLiteral( "symbology_mode" ) );
+  symbologyModeElement.appendChild( QgsXmlUtils::writeVariant( static_cast<int>( symbologyMode() ), doc ) );
+  myRootNode.appendChild( symbologyModeElement );
+
+  QDomElement symbologyScaleElement = myDocument.createElement( QStringLiteral( "symbology_scale" ) );
+  symbologyScaleElement.appendChild( QgsXmlUtils::writeVariant( symbologyScale(), doc ) );
+  myRootNode.appendChild( symbologyScaleElement );
+
+  QDomElement encodingElement = myDocument.createElement( QStringLiteral( "encoding" ) );
+  encodingElement.appendChild( QgsXmlUtils::writeVariant( encoding(), doc ) );
+  myRootNode.appendChild( encodingElement );
+
+  QDomElement crsElement = myDocument.createElement( QStringLiteral( "crs" ) );
+  crsElement.appendChild( QgsXmlUtils::writeVariant( crs(), doc ) );
+  myRootNode.appendChild( crsElement );
+
+  QDomElement mapThemeElement = myDocument.createElement( QStringLiteral( "map_theme" ) );
+  mapThemeElement.appendChild( QgsXmlUtils::writeVariant( mapTheme(), doc ) );
+  myRootNode.appendChild( mapThemeElement );
+
+  QDomElement layersElement = myDocument.createElement( QStringLiteral( "layers" ) );
```

Comment:
This isn't being restored in the loadSettingsFromXML

Reply:
Yes, that's what I meant in this comment https://github.com/gacarrillor/QGIS/pull/35#issuecomment-2007478106